ui/src/pages/backoffice/Patron/PatronDetails/components/PatronPastLoans/PatronPastLoans.js

```diff
@@ -0,0 +1,122 @@
+import { dateFormatter } from '@api/date';
+import { DocumentTitle } from '@components/Document';
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { Loader, Error, ResultsTable } from '@components';
+import { loan as loanApi } from '@api';
+import { invenioConfig } from '@config';
+import { BackOfficeRoutes } from '@routes/urls';
+import { SeeAllButton } from '@pages/backoffice/components/buttons';
+
+export default class PatronPastLoans extends Component {
+  componentDidMount() {
+    const patronPid = this.props.patronDetails.user_pid
+      ? this.props.patronDetails.user_pid
+      : null;
+    this.props.fetchPatronPastLoans(patronPid);
+  }
+
+  seeAllButton = () => {
+    const patronPid = this.props.patronDetails.user_pid;
+    const path = BackOfficeRoutes.loansListWithQuery(
+      loanApi
+        .query()
+        .withPatronPid(patronPid)
+        .withState(invenioConfig.circulation.loanRequestStates)
+        .qs()
+    );
+    return <SeeAllButton to={path} />;
+  };
+
+  viewDetails = ({ row }) => {
+    return (
+      <Link
+        to={BackOfficeRoutes.loanDetailsFor(row.metadata.pid)}
+        data-test={row.metadata.pid}
+      >
+        {row.metadata.pid}
+      </Link>
+    );
+  };
+
+  viewDocument = ({ row }) => {
+    return (
+      <Link
+        to={BackOfficeRoutes.documentDetailsFor(row.metadata.document_pid)}
+        data-test={row.metadata.pid}
+      >
+        <DocumentTitle document={row.metadata.document} />
+      </Link>
+    );
+  };
+
+  viewItem = ({ row }) => {
+    return (
+      <Link
+        to={BackOfficeRoutes.itemDetailsFor(row.metadata.item_pid)}
+        data-test={row.metadata.pid}
+      >
+        {row.metadata.item.barcode}
+      </Link>
+    );
+  };
```

Comment:
They are all the same thing `viewFunction(urlBuilder, urlParam, label)` something like that we can make it more flexible by passing params as a literal `{urlBuilder: (), pid: '..', label: '..'}` wdyt?